### DISCUSSION TITLE
Flow - Delete all items within a collection (possible?)

### DISCUSSION THREAD

**Discussion by georgecm** [directus/directus#15320] [Archive]
Hi, I have created a Flow (using the UI) to update all my collection items with the results of an HTTP request (in JSON).

I simply wanted to replace all item details based on their unique key (ex. ID) but the documentation is so limited I am not even sure it's possible.

So what I tried next, is to totally delete all items and then re-create them all together which initially seems possible, before you realise that deleting has a limit of 100 (based on testing, I didn't find any documentation).

Is this possible in a different way? If not, it should be... Thanks in advance

**Comment by br41nslug** [comment]
Im not sure you should delete all in a Flow but it is possible 😄 
![image](https://user-images.githubusercontent.com/9389634/187465853-f3a52eb5-1787-4d26-b06b-6313fa2b76d9.png)

**Reply by georgecm** [reply]
Wow thanks :) Was this somewhere documented? 

Why not? Meaning, do you see an alternative in my situation?

**Reply by br41nslug** [reply]
> Why not?

Not one specific example but more a general warning of automating workflows that delete all the records which could leave the collection empty when an error occurs.

> Meaning, do you see an alternative in my situation?
> I simply wanted to replace all item details based on their unique key (ex. ID)

Sounds like looping over the items and updating/deleting them one by one might be a more secure way to go (although a bit more work to implement). Depending on the type of data in that collection looping may be overkill the appropriate solution will vary depending on how crucial the data is.

**Reply by georgecm** [reply]
Definitely a good point regarding the errors. I could probably try to keep a "backup" before deleting everything.

My initial thought was to update the ones that need updating, but I couldn't figure out how within Flows (I'm very new to using Directus).

Thanks again!

**Reply by rijkvanzanten** [reply]
@georgecm I think you can achieve what you're looking for by using a `filter` query, so you update only that filtered set of data:

<img width="582" alt="CleanShot 2022-08-31 at 14 21 37@2x" src="https://user-images.githubusercontent.com/9141017/187751496-79f73563-f59e-41b2-9e47-8ab93f9d08fa.png">

**Reply by rijkvanzanten** [reply]
(this is in the configuration of the "Update Data" flow)

**Reply by rijkvanzanten** [reply]
> but I couldn't figure out how within Flows (I'm very new to using Directus).

No problem! Flows is very new in general, so we're also still figuring out what the most user friendly way of doing things is 🙂  (cc @erondpowell )

**Reply by imknight** [reply]
@rijkvanzanten can I check with you, how you put in array if the array coming from $trigger or $last? I have a similar query like the screen shot, but it won't work if I put "{{$last.body.keys}}"

**Reply by rijkvanzanten** [reply]
> @rijkvanzanten can I check with you, how you put in array if the array coming from $trigger or $last? I have a similar query like the screen shot, but it won't work if I put "{{$last.body.keys}}"

I think you might be running into https://github.com/directus/directus/discussions/15945#discussioncomment-3861135

**Reply by imknight** [reply]
got it, will try thanks!
